server/mneme_api_response.rb

```diff
@@ -80,6 +81,49 @@ def extractAssignment(assignment)
     assign
   end
 
+  ############ Filter assignments
+  # The filter will take a single assignment and return it if it is acceptable and return nil
+  # if the assignment should not be used.
+
+  # The checks require date comparisons. To make testing easier the actual filter is implemented as
+  # a class method which requires the time as an argument.  The following instance method defaults
+  # the time to the current time to now.  'now' is a method that can be overridden by tests.
+
+  # This is an instance method to bind the time used for filtering to the current time.
+  def filter(assignment)
+    result = self.class.filter_two(assignment, Time.now().to_i)
+    logger.debug "#{self.class.to_s}:#{__method__}: #{__LINE__}: skipping assignment: [#{assignment.inspect}]" unless result
+    return result
+  end
+
+  # This is a class method for easy testing.  The current time must be passed in
+  def self.filter_two(assignment, now)
```

Comment:
As to whether the time needed to be passed in or be assigned a default value, I was curious about whether that was possible in Ruby.  I found some examples of this:

• http://stackoverflow.com/q/16322706/543738
• http://www.skorks.com/2009/08/method-arguments-in-ruby/

I've also read that in Ruby versions 2.0 and greater, that arguments can be named in method calls.